include/s3select_functions.h

```diff
@@ -1230,79 +1231,270 @@ struct _fn_in : public base_function
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else if(i == size-1) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case ANYCHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '_' && i == size-1) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else if(*p == '%' && i == size-1) {
+        v.push_back('.');
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else  if(i == size-1) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case STAR_CHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '%' && i == size-1) {
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        st = STAR_CHAR;
+        } else if(*p == '_' && i == size-1) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else  if(i == size-1) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case ESCAPE:
+        if(i == size-1) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case END:
+        return v;
+
+      default:
+        throw base_s3select_exception("missing state!");
+        break;
     }
-    return true;
+    p++;
+    i++;
   }
-};
+  return v;
+}
+
+bool operator()(bs_stmt_vec_t* args, variable* result) override
+  {
+    auto iter = args->begin();
+
+    std::string escape_char = "\\";
+    std::string like_string;
+
+      base_statement* escape_expr = *iter;
+      iter++;
+      base_statement* like_expr = *iter;
+      iter++;
+      base_statement* main_expr = *iter;
 
+      value like_expr_val = like_expr->eval();
+      value escape_expr_val = escape_expr->eval();
+      
+      value main_expr_val = main_expr->eval();
+      if (main_expr_val.type != value::value_En_t::STRING)
+      {
+        throw base_s3select_exception("main expression must be string");
+      }
+      if (like_expr_val.type != value::value_En_t::STRING)
+      {
+        throw base_s3select_exception("like expression must be string");
+      }
+
+      if(escape_expr_val.is_null()) {
```

Comment:
the logic looks correct.

its better move it into constructor, it will improve efficiency much, 
since it will skip the dynmic_cast<..> per each new row, and also the ->eval()

in case like-string(and escape) are constant, and this condition could verify **ONCE** in constructor,
the operator will flow accordingly.

_fn_like( ... )
{
  if ( condition of constant are true ) { constant_state = true} else {constant_state = false;}
}
operator()
{
 if (constant_state == false)
{
           like_expr_val = like_expr->eval();
           escape_expr_val = escape_expr->eval();
            std::vector<char> like = transform(like_string, escape_char[0]);
            std::string str(like.begin(), like.end());
            compiled_regex = std::regex(str);
}

 if ( std::regex_match(main_expr_val.to_string(), compiled_regex)) { ..... }

}